src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -24,14 +28,16 @@ public SafeFileHandle(IntPtr preexistingHandle, bool ownsHandle) : this(ownsHand
             SetHandle(preexistingHandle);
         }
 
-        internal bool? IsAsync { get; set; }
+        public bool IsAsync { get; private set; }
```

Comment:
If code does:
```C#
var handle = new SafeFileHandle(myAsyncHandle, ownsHandle: true);
```
what causes IsAsync to be correct?

Reply:
There is nothing on Unix that indicates that given file handle was opened for async file IO.

https://github.com/dotnet/runtime/blob/05327b18be5b832089866d2326908192ae433c60/src/libraries/System.IO.FileSystem/tests/File/OpenHandle.cs#L43-L57

There is `O_ASYNC` for `open` but currently it has no effect on regular files.

We have similar situation for `O_APPEND` - when the user passes a handle that was opened with  `O_APPEND` we don't tread it as `FileMode.Append`

Reply:
Yup, sorry, I missed that this was the Unix-specific IsAsync.